app/src/main/java/com/oracle/javafx/scenebuilder/app/SceneBuilderApp.java

```diff
@@ -507,24 +533,60 @@ public void handleOpenFilesAction(List<String> files) {
         }
 
         EditorController.updateNextInitialDirectory(fileObjs.get(0));
-
+        
+        Consumer<Map<File, Exception>> onError = errors -> showFileOpenErrors(errors, 
+                                                            () -> WelcomeDialogWindowController.getInstance().getStage());
+        
         // Fix for #45
         if (userLibrary.isFirstExplorationCompleted()) {
-            performOpenFiles(fileObjs);
+            performOpenFiles(fileObjs, onError, onSuccess);
         } else {
             // open files only after the first exploration has finished
             userLibrary.firstExplorationCompletedProperty().addListener(new InvalidationListener() {
                 @Override
                 public void invalidated(Observable observable) {
                     if (userLibrary.isFirstExplorationCompleted()) {
-                        performOpenFiles(fileObjs);
```

Comment:
Previously `performOpenFiles()` was called before removing the user library listener, now it's after. Does the order make any difference?

Reply:
I have to admit that I was thinking about this as well, but eventually I found no difference. I kind of dislike how this is handled but it works. Cleaning this up could be a separate PR. At least I have not observed a difference in behavior. Hence I've kept it this way. The fileObjs list is not an observable one or so, its just plain data with no other things attached so I assumed this as saf.e